hibernate-core/src/test/java/org/hibernate/orm/test/schemavalidation/ValidateConstraintTests.java

```diff
@@ -0,0 +1,336 @@
+				.addAnnotatedClasses( Gender.class, Person1.class )
+				.buildMetadata();
+		final Metadata schema2 = new MetadataSources( registryScope.getRegistry() )
+				.addAnnotatedClasses( Gender.class, Person2.class )
+				.buildMetadata();
+		final Metadata schema3 = new MetadataSources( registryScope.getRegistry() )
+				.addAnnotatedClasses( Gender.class, Person3.class )
+				.buildMetadata();
+		final Metadata schema4 = new MetadataSources( registryScope.getRegistry() )
+				.addAnnotatedClasses( Gender.class, Person4.class )
+				.buildMetadata();
+		final Metadata schema5 = new MetadataSources( registryScope.getRegistry() )
+				.addAnnotatedClasses( Gender.class, Person5.class )
+				.buildMetadata();
+
+		final HibernateSchemaManagementTool schemaTooling = new HibernateSchemaManagementTool();
+		schemaTooling.injectServices( registry );
+
+		// create the initial schema
+		System.out.println( "> Creation" );
+		createSchema( schema1, schemaTooling, registry );
+		schemaToDrop = schema1;
+
+		// validate the first schema change
+		System.out.println( "> Migration #1" );
+		try {
+			validateSchema( schema2, schemaTooling, registry );
+			if ( validationType == ConstraintValidationType.ALL ) {
+				fail( "Expecting an exception" );
+			}
+		}
+		catch (SchemaManagementException expected) {
+			if ( validationType == ConstraintValidationType.NONE ) {
+				fail( "Not expecting an exception" );
+			}
+		}
+		updateSchema( schema2, schemaTooling, registry );
+		schemaToDrop = schema2;
+
+		// validate the second schema change
+		System.out.println( "> Migration #2" );
+		try {
+			validateSchema( schema3, schemaTooling, registry );
+			if ( validationType == ConstraintValidationType.ALL ) {
+				fail( "Expecting an exception" );
+			}
+		}
+		catch (SchemaManagementException expected) {
+			if ( validationType == ConstraintValidationType.NONE ) {
+				fail( "Not expecting an exception" );
+			}
+		}
+		updateSchema( schema3, schemaTooling, registry );
+		schemaToDrop = schema3;
+
+		// validate the 3rd schema change
+		System.out.println( "> Migration #3" );
+		try {
+			validateSchema( schema4, schemaTooling, registry );
+			if ( validationType == ConstraintValidationType.ALL ) {
+				fail( "Expecting an exception" );
+			}
+		}
+		catch (SchemaManagementException expected) {
+			if ( validationType == ConstraintValidationType.NONE ) {
+				fail( "Not expecting an exception" );
+			}
+		}
+		updateSchema( schema4, schemaTooling, registry );
+		schemaToDrop = schema4;
+
+		// validate the 4th schema change
+		System.out.println( "> Migration #4" );
+		try {
+			validateSchema( schema5, schemaTooling, registry );
+			if ( validationType == ConstraintValidationType.ALL ) {
+				fail( "Expecting an exception" );
+			}
+		}
+		catch (SchemaManagementException expected) {
+			if ( validationType == ConstraintValidationType.NONE ) {
+				fail( "Not expecting an exception" );
+			}
+		}
+		updateSchema( schema5, schemaTooling, registry );
```

Comment:
This seems to fail on Oracle because it can't match the index with the generated name `IDX6vwh1addgrj8v4h4od0bey3th` for `persons (name)` against the `person_name` index on the same column. Just wanted to give you this hint as I looked into the failure report.

Reply:
I'm not following.  You mean with the `ALL` param?  In that case, we are expecting a failure for exactly that reason.

Reply:
Oh, you mean on the schema-update?

Reply:
> Oh, you mean on the schema-update?

Yeah, correct.